src/BloomExe/web/ReadersHandler.cs

```diff
@@ -318,24 +318,25 @@ private static void OpenTextsFolder()
 
 				// if the system has wmctrl installed, use it to bring the folder to the front
 				Process.Start(new ProcessStartInfo()
-					{
-						FileName = "wmctrl",
-						Arguments = "-a \"Sample Texts\"",
-						UseShellExecute = false,
-						ErrorDialog = false // do not show a message if not successful
-					});
+				{
+					FileName = "wmctrl",
+					Arguments = "-a \"Sample Texts\"",
+					UseShellExecute = false,
+					ErrorDialog = false // do not show a message if not successful
+				});
 			}
 		}
 
 		private static void ChooseAllowedWordListFile(IRequestInfo info)
 		{
-			// Need to do this in order to show the open file dialog
-			var t = new Thread(ShowSelectAllowedWordsFileDialog);
-			t.SetApartmentState(ApartmentState.STA);
-			t.Start(info);
+			var frm = Application.OpenForms.Cast<Form>().FirstOrDefault(f => f is Shell);
+			if (frm != null)
+			{
+				frm.Invoke(new Action<IRequestInfo>(ShowSelectAllowedWordsFileDialog), info);
+			}
```

Comment:
So what will happen if there IS no form? I don't suppose it should be possible...in which case just letting Null Ref exception be thrown if it happens might be appropriate...or if it's really OK to do nothing when there isn't a main window a comment might explain this...If we crash somewhere else or mysteriously do nothing when the user issues a command that would feel worse to me than a well-defined crash in the place where something unexpected happened.